### PR TITLE
Added required sections in physics module

### PR DESCRIPTION
**Pull request by Soumi7**
# Updates sympy.physics to follow latest docstring style guide

#### References to other Issues or PRs

#19591

#### Brief description of what is fixed or changed
- Required sections in :
  - sympy/physics/hydrogen.py
  - sympy/physics/matrices.py
  - sympy/physics/paulialgebra.py
  - sympy/physics/pring.py
  - sympy/physics/qho_1d.py
  - sympy/physics/sho.py
  - sympy/physics/secondquant.py
  - sympy/physics/wigner.py
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Soumi7** [comment]
@aktech

File: sympy/physics/hydrogen.py [hunk 1]
```diff
@@ -177,15 +183,18 @@ def E_nl_dirac(n, l, spin_up=True, Z=1, c=Float("137.035999037")):
...
+    spin_up :
         True if the electron spin is up (default), otherwise down
-    Z
+    Z :
         atomic number (1 for Hydrogen, 2 for Helium, ...)
-    c
+    c :
         speed of light in atomic units. Default value is 137.035999037,
-        taken from: http://arxiv.org/abs/1012.3627
+        taken from [1].
```
Comment (aktech) [inline sympy/physics/hydrogen.py:197]: Have you checked, if this is rendered correctly?

File: sympy/physics/pring.py [hunk 2]
```diff
@@ -7,9 +7,12 @@ def wavefunction(n, x):
     """
     Returns the wavefunction for particle on ring.
 
-    n is the quantum number, x is the angle,
-    here n can be positive as well as negative
-    which can be used to describe the direction of motion of particle
+    Explanation
+    ===========
+
+    ``n`` is the quantum number, ``x`` is the angle,
```
Comment (aktech) [inline sympy/physics/pring.py:13]: Would be good to add these as Parameters? Is this function not in public API?

File: sympy/physics/pring.py [hunk 3]
```diff
@@ -45,11 +48,19 @@ def wavefunction(n, x):
...
-    here n is the quantum number, m is the mass of the particle
-    and r is the radius of circle.
+
+    Parameters
+    ==========
+
+    ``n`` :
+        The quantum number.
+    ``m`` :
+        The mass of the particle.
```
Comment (aktech) [inline sympy/physics/pring.py:61]: The mass of the particle -> Mass of the particle

File: sympy/physics/qho_1d.py [hunk 4]
```diff
@@ -67,12 +76,15 @@ def E_n(n, omega):
 def coherent_state(n, alpha):
     """
     Returns <n|alpha> for the coherent states of 1D harmonic oscillator.
-    See https://en.wikipedia.org/wiki/Coherent_states
+    See https://en.wikipedia.org/wiki/Coherent_states.
```
Comment (aktech) [inline sympy/physics/qho_1d.py:79]: Probably not a good idea to add full stop after a url, people end up copying the url with link and it won't work then.

File: sympy/physics/secondquant.py [hunk 5]
```diff
@@ -1021,11 +1069,17 @@ def up(self, i):
         """
         Performs the action of a creation operator.
 
+        Explanation
+        ===========
+
         If below fermi we try to remove a hole,
-        if above fermi we try to create a particle.
+        If above fermi we try to create a particle.
```
Comment (aktech) [inline sympy/physics/secondquant.py:1076]: I think this doesn't needs to capitalise as there is a comma right before it.

File: sympy/physics/secondquant.py [hunk 6]
```diff
@@ -1021,11 +1069,17 @@ def up(self, i):
...
 
-        if general index p we return Kronecker(p,i)*self
-        where i is a new symbol with restriction above or below.
+        If general index p we return Kronecker(p,i)*self
```
Comment (aktech) [inline sympy/physics/secondquant.py:1078]: back-ticks for Kronecker(p,i)*self

File: sympy/physics/secondquant.py [hunk 7]
```diff
@@ -1161,7 +1221,7 @@ def _add_orbit(self, i):
     @classmethod
     def _count_holes(cls, list):
         """
-        returns number of identified hole states in list.
+        Returns number of identified hole states in list.
```
Comment (aktech) [inline sympy/physics/secondquant.py:1224]: Returns number -> Returns the number

File: sympy/physics/wigner.py [hunk 8]
```diff
@@ -28,13 +28,14 @@
...
 Credits and Copyright
-~~~~~~~~~~~~~~~~~~~~~
+=====================
 
 This code was taken from Sage with the permission of all authors:
 
 https://groups.google.com/forum/#!topic/sage-devel/M4NZdu-7O38
 
-AUTHORS:
+AUTHORS
```
Comment (aktech) [inline sympy/physics/wigner.py:37]: The capitalisation here should be same as: https://github.com/sympy/sympy/pull/20413/files#diff-d589b76690102a574278027b9dbf6f19a9918f683058dc955c5962ebd77a8005R174

File: sympy/physics/wigner.py [hunk 9]
```diff
@@ -276,21 +294,25 @@ def _big_delta_coeff(aa, bb, cc, prec=None):
...
-    INPUT:
-
-    -  ``aa`` - first angular momentum, integer or half integer
-
-    -  ``bb`` - second angular momentum, integer or half integer
+    Parameters
+    ==========
 
-    -  ``cc`` - third angular momentum, integer or half integer
+    aa :
```
Comment (aktech) [inline sympy/physics/wigner.py:300]: back-ticks around these parameters

File: sympy/physics/pring.py (hunk 3 above)
Comment (Soumi7) [reply sympy/physics/pring.py:61]: Changed.

File: sympy/physics/pring.py (hunk 2 above)
Comment (Soumi7) [reply sympy/physics/pring.py:13]: Added these as parameters.

File: sympy/physics/hydrogen.py (hunk 1 above)
Comment (Soumi7) [reply sympy/physics/hydrogen.py:197]: Its not getting rendered correctly. I replaced it by its original link.

File: sympy/physics/qho_1d.py (hunk 4 above)
Comment (Soumi7) [reply sympy/physics/qho_1d.py:79]: Removed the full stop,

File: sympy/physics/secondquant.py (hunk 6 above)
Comment (Soumi7) [reply sympy/physics/secondquant.py:1078]: Added them.

File: sympy/physics/wigner.py (hunk 9 above)
Comment (Soumi7) [reply sympy/physics/wigner.py:300]: Changed.

File: sympy/physics/wigner.py (hunk 8 above)
Comment (Soumi7) [reply sympy/physics/wigner.py]: Changed.

File: sympy/physics/secondquant.py (hunk 7 above)
Comment (Soumi7) [reply sympy/physics/secondquant.py]: Changed.

File: sympy/physics/secondquant.py (hunk 5 above)
Comment (Soumi7) [reply sympy/physics/secondquant.py]: Changed.

File: sympy/physics/hydrogen.py (hunk 1 above)
Comment (Soumi7) [reply sympy/physics/hydrogen.py]: Resolved.

File: sympy/physics/hydrogen.py [hunk 10]
```diff
@@ -8,11 +8,14 @@ def R_nl(n, l, r, Z=1):
...
 
-    n, l
-        quantum numbers 'n' and 'l'
-    r
+    Parameters
+    ==========
+
+    n, l :
+        Quantum numbers 'n' and 'l'.
+    r :
```
Comment (oscarbenjamin) [inline sympy/physics/hydrogen.py:16]: Maybe this should say that the parameters should be expressions representing integers. Something like:
```
n, l: Expr
    Expressions representing the quantum numbers 'n' and 'l' which should be integers in the range ...
```

**Comment by oscarbenjamin** [comment]
I left a comment which also applies to other parts of the diff. Otherwise this looks good to me.

File: sympy/physics/hydrogen.py (hunk 10 above)
Comment (Soumi7) [reply sympy/physics/hydrogen.py:17]: @aktech I have added a similar description here, but I am not sure about the range of integers. Could you look into this?

Comment (aktech) [reply sympy/physics/hydrogen.py:17]: As per my research, the range is following
q-20numbers

[Reference link](https://courses.lumenlearning.com/introchem/chapter/quantum-numbers/)

Also, I think `n` and `l` should be on different lines.

File: sympy/physics/hydrogen.py [hunk 11]
```diff
@@ -87,15 +91,18 @@ def Psi_nlm(n, l, m, r, phi, theta, Z=1):
     Returns the Hydrogen wave function psi_{nlm}. It's the product of
     the radial wavefunction R_{nl} and the spherical harmonic Y_{l}^{m}.
 
-    n, l, m
+    Parameters
+    ==========
+
+    n, l, m :
```
Comment (aktech) [inline sympy/physics/hydrogen.py:97]: as mentioned above, I believe this should be on different lines.

Comment (Soumi7) [reply sympy/physics/hydrogen.py]: Changed.

File: sympy/physics/hydrogen.py (hunk 10 above)
Comment (Soumi7) [reply sympy/physics/hydrogen.py:18]: Thanks! Have added them.

File: sympy/physics/hydrogen.py [hunk 12]
```diff
@@ -87,15 +92,22 @@ def Psi_nlm(n, l, m, r, phi, theta, Z=1):
     Returns the Hydrogen wave function psi_{nlm}. It's the product of
     the radial wavefunction R_{nl} and the spherical harmonic Y_{l}^{m}.
 
-    n, l, m
-        quantum numbers 'n', 'l' and 'm'
-    r
+    Parameters
+    ==========
+
+    n : integer ranging from 1 to infinity.
```
Comment (aktech) [inline sympy/physics/hydrogen.py:98]: I would suggest something like: **n : integer with possible values as 1, 2, 3, 4,..**.
The reason being, infinity is not in the domain of integers, which makes the statement "1 to infinity." misleading.

Applies for all such instances below as well.

Comment (Soumi7) [reply sympy/physics/hydrogen.py]: Corrected them.

**Review by Sc0rpi0n101 — APPROVED** [review]
This looks good to me, except a minor comment.

File: sympy/physics/hydrogen.py [hunk 13]
```diff
@@ -87,15 +92,22 @@ def Psi_nlm(n, l, m, r, phi, theta, Z=1):
...
     the radial wavefunction R_{nl} and the spherical harmonic Y_{l}^{m}.
 
-    n, l, m
-        quantum numbers 'n', 'l' and 'm'
-    r
+    Parameters
+    ==========
+
+    n : integer with possible values as 1, 2, 3, 4,...
+        Principal Quantum Number.
```
[the remark this answers is not in the document]
Comment (Soumi7) [reply sympy/physics/hydrogen.py]: Yes, I made the changes.

**Pull request merged**